src/decision/justification_heuristic.h

```diff
@@ -38,7 +38,15 @@
 namespace CVC4 {
 namespace decision {
 
-class JustificationHeuristic : public ITEDecisionStrategy {
+template <bool ref_count>
+class JustificationHeuristic;
+
+template <bool check2>
```

Comment:
`check2` is not very descriptive and sounds a bit like there should be a `check1`, maybe something like `checkInPairs`? Maybe someone has an even better idea.